fastly/resource_fastly_tls_subscription.go

```diff
@@ -294,10 +294,37 @@ func resourceFastlyTLSSubscriptionRead(_ context.Context, d *schema.ResourceData
 	if err != nil {
 		return diag.FromErr(err)
 	}
+
+	// NOTE: The configuration_id should change depending on scenario.
+	// For example, prior to a subscription renewal the ID is the same.
+	// Once a subscription is renewed, we need to search for the latest ID.
+	// The following API endpoint is used to search for the latest ID.
+	// https://www.fastly.com/documentation/reference/api/tls/custom-certs/domains/#list-tls-domains
 	err = d.Set("configuration_id", subscription.Configuration.ID)
 	if err != nil {
 		return diag.FromErr(err)
 	}
+	var tlsDomains []*gofastly.TLSDomain
+	tlsDomains, err = conn.ListTLSDomains(&gofastly.ListTLSDomainsInput{
+		FilterTLSCertificateID: certificateID,
+		Include:                "tls_activations",
+		Sort:                   "tls_activations.created_at",
+	})
+	for _, tlsDomain := range tlsDomains {
+		// Activations may be empty (omitempty)
+		if tlsDomain.Activations == nil {
+			break
+		}
+		activations := len(tlsDomain.Activations)
+		if activations > 0 {
+			err = d.Set("configuration_id", tlsDomain.Activations[activations-1].Configuration.ID)
```

Comment:
Since we're sorting by created at, should we change this always be `tlsDomain.Activations[0]` ?

Effectively, this list is always of length 1 as noted above in the comments.

Reply:
I made a similar suggestion and the TLS team suggested that there could be (in _very_ specific cases) be more than one entries and so it's safer to sort and grab appropriate end

Reply:
Updated!